### ISSUE TITLE
Alby should ask in which account I want to login

### ISSUE THREAD

**Issue by bitcoinuser** [getAlby/lightning-browser-extension#833]
Hi,

I have two accounts on Alby, so I disconnected from stacker.news service. then I clicked to login again on stacker.news. 

What I expected:
Alby ask in which from the two accounts I would like to login.

What happened:
Alby logged in automatically using the default Alby account

**Comment by bitcoinuser** [comment]
> It's very useful to choose an account. I'll try this optimization and improvement

Maybe you think this interesting too: https://github.com/getAlby/lightning-browser-extension/issues/1853

**Comment by qqqzhch** [comment]
> > It's very useful to choose an account. I'll try this optimization and improvement
> 
> Maybe you think this interesting too: #1853
Yes. Let me take a look at the logics of login and account first. 
Time passes quickly. I remember the first time I wrote issues https://github.com/getAlby/lightning-browser-extension/issues/435  It's about login

**Comment by qqqzhch** [comment]
Added an account selection component, which is used in the authorization page, and the effect is as follows, and the style code and test code are still under development
![#](https://qqqzhch.github.io/Albyimage/WX20221212-111023.png)
![#](https://qqqzhch.github.io/Albyimage/WX20221212-111114.png)
![#](https://qqqzhch.github.io/Albyimage/WX20221212-111241.png)
![#](https://qqqzhch.github.io/Albyimage/WX20221212-111306.png)

**Comment by qqqzhch** [comment]
Refresh the page after logging in, and the page judges whether there is a link to the wallet.

Still execute window.webln.enable()?

**Comment by qqqzhch** [comment]
I retested again

The wallet has two accounts lnd1 and lnd2
1 wallet default account lnd1
   The webpage login gets the account of lnd1
2 Switch the wallet default account to lnd2
3 Web page logout
4 Click to log in on the webpage, there is no authorization dialog box, and the account of lnd2 is obtained

I think the second login, without the authorization dialog box, should get the account of lnd1

I'm going to try to see if I can get the account of lnd1

**Comment by reneaaron** [comment]
> I think the second login, without the authorization dialog box, should get the account of lnd1

So you would save where the user has logged in before and use that account? What if a user has multiple accounts that he is using on one website? 

I think it would make sense to discuss the concept before you start to implement this.

**Comment by qqqzhch** [comment]
I read a part of the code, but I haven't read it completely. think of a way first
window.webln.enable
`const allowance = await db.allowances
    .where("host")
    .equalsIgnoreCase(host)
    .first();`

window.webln.lnurl
`if (lnurlDetails.tag !== "login")
    throw new Error(
      "LNURL-AUTH FAIL: incorrect tag: ${lnurlDetails.tag} was used"
    );
  const connector = await state.getState().getConnector();
`



Allowance is looking for the first
Can I find a first connector according to the allowance and find an authorized connector 
 the connector is the first one according to the allowance

If a site need links multiple connector，Add new components in the wallet to link the account with the site，

lookup logic
First find the connector with allowances,
If the found connectors contains the default connector of the wallet, return  the default connector
If not contained, return the first

**Comment by qqqzhch** [comment]
The current wallet allowance is not specific to the account. There will be multiple accounts in the wallet, and the amount of each account is different. I suggest adding an account ID when recording allowance information
`
          await db.allowances.add({
            host: host,
            name: message.origin.name,
            imageURL: message.origin.icon,
            enabled: true,
            lastPaymentAt: 0,
            totalBudget: 0,
            remainingBudget: 0,
            createdAt: Date.now().toString(),
            lnurlAuth: false,
            tag: "",
          })
`

**Comment by qqqzhch** [comment]
Do we plan to save or cache multiple Connectors @escapedcat

**Comment by escapedcat** [comment]
> Do we plan to save or cache multiple Connectors

I don't understand. Can you explain what you mean?

**Comment by fczuardi** [comment]
I think the expectation of the original reporter, with logging-out from [stacker.news](https://stacker.news), was that using the login button again would display the Alby popup again (where he could choose an account). However since  the toggle inside the Allowances>Edit, that is labeled:

> Enable website login / Automatically log in without confirmation when the website requests.

Has the default value of "ON", then the second login will always try to reuse the account from the last login.

**Comment by fczuardi** [comment]
Here is a video reproducing the original report:

[![](https://i9.ytimg.com/vi/tdq-qjr3jUI/mqdefault.jpg?sqp=CMC-rKIG-oaymwEmCMACELQB8quKqQMa8AEB-AH-CYAC0AWKAgwIABABGGUgZShlMA8=&rs=AOn4CLCP5IfGIHU_akxMXOuvgT7Uv9inGg)](https://youtu.be/tdq-qjr3jUI)
https://youtu.be/tdq-qjr3jUI

And here is a video with a workaround:
[![](https://i9.ytimg.com/vi/CK4EAFw9A_I/mq1.jpg?sqp=COzArKIG-oaymwEmCMACELQB8quKqQMa8AEB-AHUBoAC4AOKAgwIABABGGUgZShlMA8=&rs=AOn4CLDU4AhQEYloUbbhcptFsm0NRDg4Aw)](https://youtu.be/CK4EAFw9A_I)
https://youtu.be/CK4EAFw9A_I
